File: day2.py
#!/usr/bin/env python3

import sys


def main():
    h = 0
    d = 0
    d2 = 0
    a = 0
    with open(sys.argv[1], 'r') as fhan:
        for line in fhan.readlines():
            [move, delta] = line.split(' ')
            if move == 'forward':
                h += int(delta)
                d2 += a * int(delta)
            elif move == 'down':
                d += int(delta)
                a += int(delta)
            elif move == 'up':
                d -= int(delta)
                a -= int(delta)
            else:
                print('Unexpected input:', line)
    print('Part 1:', h * d)
    print('Part 2:', h * d2)


if __name__ == '__main__':
    main()

File: day2.rb
h = 0
d = 0
d2 = 0
a = 0

File.open(ARGV[0]).each do |line|
  move, dist = line.split(' ')
  dist = dist.to_i
  if move == 'forward'
    h += dist
    d2 += a * dist
  elsif move == 'down'
    d += dist
    a += dist
  elsif move == 'up'
    d -= dist
    a -= dist
  end
end

puts("Part 1: #{h * d}")
puts("Part 2: #{h * d2}")
